src/components/PupilViews/PupilQuiz/PupilQuiz.view.tsx

```diff
@@ -1,28 +1,93 @@
-import { OakFlex } from "@oak-academy/oak-components";
+import {
+  OakBackLink,
+  OakFlex,
+  OakLessonBottomNav,
+  OakLessonLayout,
+  OakLessonTopNav,
+  OakPrimaryButton,
+} from "@oak-academy/oak-components";
 
 import {
   QuestionsArray,
   QuizEngineProvider,
+  useQuizEngineContext,
 } from "@/components/PupilComponents/QuizEngineProvider";
 import { QuizRenderer } from "@/components/PupilComponents/QuizRenderer";
+import { useLessonEngineContext } from "@/components/PupilComponents/LessonEngineProvider";
 
 type PupilViewsQuizProps = {
   questionsArray: QuestionsArray;
 };
 
-export const PupilViewsQuiz = ({ questionsArray }: PupilViewsQuizProps) => {
+type CurrentSection = "starter-quiz" | "exit-quiz";
+
+const QuizInner = () => {
+  const { currentSection, updateCurrentSection } = useLessonEngineContext();
+  const quizEngineContext = useQuizEngineContext();
+
+  const { currentQuestionIndex, questionState, handleNextQuestion } =
+    quizEngineContext;
+
+  const isFeedbackMode =
+    questionState[currentQuestionIndex]?.mode === "feedback";
+
   return (
-    <QuizEngineProvider questionsArray={questionsArray}>
-      <OakFlex
-        $width={"100vw"}
-        $height={"100vh"}
-        $background={"bg-decorative1–main"}
-        $flexDirection={"column"}
-        $alignItems={"center"}
-        $pt="inner-padding-xl"
-      >
+    <OakLessonLayout
+      bottomNavSlot={
+        <OakLessonBottomNav>
```

Comment:
I feel this is pushing things in terms of readability . Could you pop this in its own component (can be inside this file)